FIAT/functional.py

```diff
@@ -71,39 +71,18 @@ def __init__(self, ref_el, target_shape,
         return
 
     def evaluate(self, f):
-        """Evaluates the functional on some callable object f."""
-        result = 0
-
-        # non-derivative part
-        # TODO pt_dict? comp?
-        for pt in pt_dict:
-            wc_list = pt_dict[pt]
-            for (w, c) in wc_list:
-                if comp == tuple:
-                    result += w * f(pt)
-                else:
-                    result += w * f(pt)[comp]
-
-        # Import AD modules from ScientificPython
-        # import Scientific.Functions.Derivatives as Derivatives
-        for pt in self.deriv_dict:
-            dpt = tuple([Derivatives.DerivVar(pt[i], i, self.max_deriv_order)
-                         for i in range(len(pt))
-                         ])
-            for (w, a, c) in self.deriv_dict[pt]:
-                fpt = f(dpt)
-                order = sum(a)
-                if c == tuple():
-                    val_cur = fpt[order]
-                else:
-                    val_cur = fpt[c][order]
-                for i in range(len[a]):
-                    for j in range(a[j]):
-                        val_cur = val_cur[i]
-
-                result += val_cur
+        """Obsolete and broken functional evaluation.
 
-        return result
+        To evaluate the functional, call it on the target function:
+
+          functional(function)
+        """
+
+        raise AttributeError("To evaluate the functional just call it on a function.")
```

Comment:
Functionals are functions. How do to you call a function? f(x).

Now, usually one would not change interfaces like that, in order to avoid breaking other people's code. However, in this case we know that nobody can possibly have been using what was there so there is no reason not to fix the interface.